tox.ini

```diff
@@ -1,74 +1,45 @@
 [tox]
+# generative list of environments (matrix) to run by default
 envlist =
-    clean,
-    py27-d1.7, py26-d1.8,
-    py33-d1.7, py33-d1.8,
-    py34-d1.7, py34-d1.8,
-    pypy-d1.7, pypy-d1.8,
-    pypy3-d1.7, pypy3-d1.8,
-    report
-[testenv]
-commands= coverage run -m unittest discover
-
-[testenv:clean]
-deps=
-  coverage
-commands=
-  coverage erase
-
-[testenv:report]
-deps=
-  coverage
-commands=
-  coverage report --include "cbs/*"
-  coverage html --include "cbs/*"
-
-# Python 2.7
-
-[testenv:py27-d1.7]
-deps = django>=1.7,<1.8
-  coverage
-
-[testenv:py27-d1.8]
-deps = django>=1.8,<1.9
-  coverage
+  clean,
+  py{27,33,34,35,pypy,pypy3}-django{1.6,1.7,1.8,1.9},
```

Comment:
Tox has no facility to fail on bad interpreter factors, I guess factor namespaces would be a nice addition to tox to prevent this sort of problem.

If `/` was the namespace separator and the `tox/intepreter` namespace had validation, then an example would be thus:
```ini
[tox]
    envlist=tox/interpreter/{py27,py33,py34,py35,pypy,pypy3}-django{16,17,18,19}
```